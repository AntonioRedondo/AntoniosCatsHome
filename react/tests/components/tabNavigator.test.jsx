import React from "react";
import { shallow } from "enzyme";

import TabNavigator from "../../src/js/components/tabNavigator.jsx";


describe("Given the TabNavigator component", () => {
	const mockProps = {
		location: { pathname: "/how-to-adopt" }
	};
	
	describe("When the component is mounted", () => {
		it("Should render children", () => {
			const wrapper = shallow(
				<TabNavigator { ...mockProps } />
			);
			
			expect(wrapper.name()).toBe("div");
			expect(wrapper.childAt(0).name()).toBe("nav");
			expect(wrapper.find("NavLink").length).toBe(2);
			expect(wrapper.childAt(1).name()).toBe("CSSTransitionGroup");
			expect(wrapper.childAt(1).childAt(0).name()).toBe("FadeElement");
			expect(wrapper.find("FadeElement").prop("pathname")).toBe("/how-to-adopt");
		});
	});
	
});
