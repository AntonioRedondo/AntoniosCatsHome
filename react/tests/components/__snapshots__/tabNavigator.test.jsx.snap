// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the TabNavigator component When the component is mounted Should render children 1`] = `
<div>
  <styled.nav>
    <ul>
      <li>
        <Styled(NavLink)
          activeClassName="tab-navigator--active"
          exact={true}
          to="/"
        >
          Cats
        </Styled(NavLink)>
      </li>
      <li>
        <Styled(NavLink)
          activeClassName="tab-navigator--active"
          className="tab-navigator"
          to="/how-to-adopt"
        >
          How to adopt
        </Styled(NavLink)>
      </li>
    </ul>
  </styled.nav>
  <Styled(TransitionGroup)>
    <CSSTransition
      classNames="navigation-loader"
      timeout={200}
    >
      <HowToAdopt />
    </CSSTransition>
  </Styled(TransitionGroup)>
</div>
`;
